### PR TITLE
switch from Target. to instance target.

### PR DESCRIPTION
**Pull request by WalterBright**
The idea is to turn all the individual `Target` globals into a single `target` instance, which then can be passed by reference. Follow-on PRs will convert the globals to fields. This PR creates the `target` global and then does a global search/replace.

### CODE REVIEW

File: src/dmd/target.d [hunk 1]
```diff
@@ -753,3 +753,5 @@ struct Target
         }
     }
 }
+
+__gshared Target target;
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/target.d]: yes, as a follow-on.

**Comment by thewilsonator** [comment]
cc @ibuclaw @kinke

**Comment by thewilsonator** [comment]
`../../src/dmd/frontend.d(70,5): Error: undefined identifier target`

**Comment by WalterBright** [comment]
All these little pickups is why this is a separate PR :-)

File: src/dmd/target.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (thewilsonator) [reply src/dmd/target.d]: I suspect it will be easier to do the review all at once.

[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/target.d]: ok

[the remark this answers is not in the document]
Comment (thewilsonator) [reply src/dmd/target.d]: I agree in general, but this needs to wait for input from LDC and GDC. I suspect waiting twice will be slower.

**Comment by WalterBright** [comment]
The follow-on shouldn't need their input. It's part of the point of this change.

File: src/dmd/target.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/target.d]: The follow-on shouldn't need their input. It's part of the point of this change.

**Comment by ibuclaw** [comment]
> which then can be passed by reference.

Any examples?

**Comment by WalterBright** [comment]
> Any examples?

https://github.com/dlang/dmd/pull/9057/files#diff-12efbb2a88b713dbb0fd0a64673ba132R2102

**Comment by WalterBright** [comment]
Accessing mutable globals is a code smell:

1. can't make functions pure
2. threading issues
3. side channel side effects
4. undocumented dependencies
5. harder to understand / modify code without breaking things
6. ugly code is generated for PIC access to globals

While dmd is probably too far gone to eliminate such, minimizing their use makes it better.

File: src/dmd/target.d (hunk 1 above)
Comment (jacob-carlborg) [inline src/dmd/target.d]: `extern (C++)`?

**Comment by thewilsonator** [comment]
LGTM adding `72h ->merge`

**Comment by thewilsonator** [comment]
Bummer, oh well.

**Comment by WalterBright** [comment]
Time flies :-)

**Comment by thewilsonator** [comment]
Thats what happens when you sneeze when merge PRs, don't sneeze and merge, its bad mmmkay.

**Pull request merged**